src/audio/dai.c

```diff
@@ -217,6 +217,57 @@ static void dai_free(struct comp_dev *dev)
 	rfree(dev);
 }
 
+static int dai_comp_get_hw_params(struct comp_dev *dev,
+				  struct sof_ipc_stream_params *params)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	trace_dai("dai_hw_params()");
+
+	/* fetching hw dai stream params */
+	ret = dai_get_hw_params(dd->dai, params);
+	if (ret < 0) {
+		trace_dai_error("dai_comp_get_hw_params() error: dai_get_hw_params failed");
+		return ret;
+	}
+
+	return 0;
+}
+
+static int dai_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
+	struct sof_ipc_stream_params hw_params;
+
+	dai_comp_get_hw_params(dev, &hw_params);
+
+	/* checks whether pcm parameters match hardware DAI parameter set
+	 * during dai_set_config(). If hardware parameter is equal to 0, it
+	 * means that it can vary, so any value is acceptable. We do not check
+	 * format parameter, because DAI is able to change format using
+	 * pcm_converter functions.
+	 */
+	if (hw_params.rate != params->rate && hw_params.rate) {
```

Comment:
Nitpick: should we first verify hw_params.rate (that is, if a fixed rate is even requested)?

Reply:
I've updated it.